src/libstore/globals.hh

```diff
@@ -12,6 +13,8 @@
 
 namespace nix {
 
+std::string listLogFormats();
```

Comment:
I don't really like that this is outside the `Settings` class, but putting it anywhere inside gives undefined references errors (due to its use in the `logFormat` description). Happy to hear alternatives.

Reply:
I think you're allowed to make it a static member of the `Settings` class if you want to

Reply:
With @LnL7's help, I created a new type for `LogFormatSetting`, which lets me remove `listLogFormats` from publicity and instead just call it in the constructor for that setting.

(Your `static` idea didn't seem to work.)